kratos/mpi/python/kratos_mpi_python.cpp

```diff
@@ -32,6 +33,12 @@ void InitializeMPIParallelRun()
 {
     // Define the World DataCommunicator as a wrapper for MPI_COMM_WORLD and make it the default.
     ParallelEnvironment::RegisterDataCommunicator("World", MPIDataCommunicator::Create(MPI_COMM_WORLD), ParallelEnvironment::MakeDefault);
+    // Register the MPICommunicator to be used as factory for the communicator.
+    ParallelEnvironment::RegisterCommunicatorFactory([](ModelPart& rModelPart, const std::string& rDataCommunicatorName)->Communicator::UniquePointer{
+        KRATOS_ERROR_IF_NOT(ParallelEnvironment::HasDataCommunicator(rDataCommunicatorName)) << "Asking for an unregistered " << rDataCommunicatorName <<  " data communicator." << std::endl;
+        const auto& r_data_communicator = ParallelEnvironment::GetDataCommunicator(rDataCommunicatorName);
```

Comment:
hm still not sure abt this
if I pass `serial` then I get an `MPICommunicator` with a (serial) `DataCommunicator` internally!

Reply:
You're right, as the `serial` is always registered. Maybe we should also check `r_data_communicator.IsDistributed()`?

Reply:
we can and it makes sense to me, but then the code that runs in serial (i.e. when someone hardocdes `serial`) will fail in parallel
not necessarily a bad thing, but pointing it out

Reply:
I think this would be good (indeed now I remember that we discussed about throwing an error @roigcarlo)

Reply:
Yep I would throw an error if it does not exists, at least in debug

Reply:
Done.